### ISSUE TITLE
Training on M3 MacBook

### ISSUE THREAD

**Issue by david-waterworth** [UKPLab/sentence-transformers#2699]
I tried to train a sentence-transformers model using the latest release (3.0.0) on my M3 MacBook. 

The problem I ran into was this line results in an `Accelerator` instance constructed with (mostly) default arguments https://github.com/huggingface/transformers/blob/f8e6ba454c6b534d6613f62199c007743dda4761/src/transformers/trainer.py#L403 

https://github.com/huggingface/transformers/blob/f8e6ba454c6b534d6613f62199c007743dda4761/src/transformers/trainer.py#L4578

(in particular despite sentence-transformers having an fp16 argument that I set to False, the default behavior of `Accelerator(**args)` seemed to set it to true with no way I see of changing that.)

This results in an exception at

https://github.com/huggingface/accelerate/blob/b24a0ef5dbb4e0408074f92f84683608eab474af/src/accelerate/accelerator.py#L461

(since device type is 'mps')

I tried uninstalling accelerate but sentence transformers requires it - I don't think the huggingface trainer does. I think there are issues with huggingface transformers not passing the  training args to the `Accelerator` constructor but also until accelerate works for mac silicon.

Do you think there's any point removing the accelerate requirement from sentence-transformers train mode? Or should I just raise this with huggingface and wait for macos accelerate support?

**Comment by david-waterworth** [comment]
I found the `use_mps` flag which I think should fix this - it's not but I think that's a torch issue (the torch wheel I've installed appears not to be built with mps enabled) but regardless this isn't a sentence-transformer issue

**Comment by tomaarsen** [comment]
Thanks for sharing! It might help some others in the future :)

- Tom Aarsen

**Comment by david-waterworth** [comment]
@tomaarsen FYI I raised an issue with torch as I ran into unimplemented methods in the mps backend. Interestingly it's appears to only be the `Cached` losses that are exercising that particular path - see https://github.com/pytorch/pytorch/issues/127676#issuecomment-2146029982

**Comment by AndreasThinks** [comment]
Ran into this issue today as well! Swapping out to a non-cached evaluator fixes it in the short term.
